server/sheet.go

```diff
@@ -239,3 +460,82 @@ func (s *Server) composeSheetRelationship(ctx context.Context, raw *api.SheetRaw
 
 	return sheet, nil
 }
+
+// SheetInfo represents the sheet related information from sheetPathTemplate.
+type SheetInfo struct {
+	EnvironmentName string
+	DatabaseName    string
+	SheetName       string
+}
+
+// parseSheetInfo matches sheetPath against sheetPathTemplate. If sheetPath matches, then it will derive SheetInfo from the sheetPath.
+// Both sheetPath and sheetPathTemplate are the full file path(including the base directory) of the repository.
+func parseSheetInfo(sheetPath string, sheetPathTemplate string) (*SheetInfo, error) {
+	placeholderList := []string{
+		"ENV_NAME",
+		"DB_NAME",
+		"NAME",
+	}
+	sheetPathRegex := sheetPathTemplate
+	for _, placeholder := range placeholderList {
+		sheetPathRegex = strings.ReplaceAll(sheetPathRegex, fmt.Sprintf("{{%s}}", placeholder), fmt.Sprintf("(?P<%s>[a-zA-Z0-9\\+\\-\\=\\_\\#\\!\\$\\. ]+)", placeholder))
+	}
+	myRegex, err := regexp.Compile(fmt.Sprintf("^%s$", sheetPathRegex))
+	if err != nil {
+		return nil, fmt.Errorf("invalid sheet path template: %q", sheetPathTemplate)
+	}
+	if !myRegex.MatchString(sheetPath) {
+		return nil, fmt.Errorf("sheet path %q does not match sheet path template %q", sheetPath, sheetPathTemplate)
+	}
+
+	sheetInfo := &SheetInfo{}
+	matchList := myRegex.FindStringSubmatch(sheetPath)
+	for _, placeholder := range placeholderList {
+		index := myRegex.SubexpIndex(placeholder)
+		if index >= 0 {
+			switch placeholder {
+			case "ENV_NAME":
+				sheetInfo.EnvironmentName = matchList[index]
+			case "DB_NAME":
+				sheetInfo.DatabaseName = matchList[index]
+			case "NAME":
+				sheetInfo.SheetName = matchList[index]
+			}
+		}
+	}
+
+	if sheetInfo.SheetName == "" {
+		return nil, fmt.Errorf("sheet name cannot be empty from sheet path %q and template %q", sheetPath, sheetPathTemplate)
+	}
+
+	return sheetInfo, nil
+}
+
+// parseBasePathFromTemplate gets the base path from template.
+func parseBasePathFromTemplate(sheetPathTemplate string) string {
```

Comment:
Maybe we should do the template checker in project api/frontend. So I use `filepath.Dir` instead of `parseBasePathFromTemplate` to get the base path from template string.